api/docs/feature-development/model.md

```diff
@@ -0,0 +1,348 @@
+
+2. We want to allow `users` to be able to leave `comments` on an `article`, so we know the article will `have_many` `comments`.
+   1. When the `author` `destroy`s the `article` then the `comments` should be `destroy`ed as well.
+3. We want to allow `users` to be able to `favorite` an `article`, so we know the `article` will `have_many` `favorites`.
+   1. When the `author` deletes the `article` then the `favorites` should be `destroy`ed as well.
+4. As an `author`, we want to be able to add multiple `tags` to an `article` and we can do that `through` `taggings`.
+   1. When the `author` delets the `aritcle` then the `taggings` should be `destroy`ed as well.
+5. We already talked about this in 4.
+
+Phew! that was a lot, lets go ahead and update the article to have the correct associations.
+
+```rb
+# frozen_string_literal: true
+
+class Article < ApplicationRecord
+  # ...
+  belongs_to :author, class_name: 'User', validate: true
+  has_many :comments, dependent: :destroy
+  has_many :favorites, dependent: :destroy
+  has_many :taggings, dependent: :destroy
+  has_many :tags, through: :taggings
+end
+```
+
+And now our tests pass!
+
+There are a few more features we'd like to add to our Article, let's talk about them.
+
+1. We'd like to get all the articles with specific tags.
+2. We'd like to get all the articles of authors a user is following.
+
+#### Tagged with
+
+```rb
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Article, type: :model do
+  describe '.tagged_with' do
+    before(:each) do
+      create_list(:tag, 3)
+      create_list(:article, 3, author: build(:author), tags: [])
+      create_list(:article, 3, author: build(:author), tags: Tag.all)
+    end
+
+    subject { described_class.tagged_with(Tag.all) }
+
+    it { expect(subject.length).to be 3 }
+  end
+end
+
+```
+
+So here, we've created 6 articles, 3 of which have tags.
+
+Our expectation is that we should only get the 3 tagged articles back.
+
+Let's make that test pass!
+
+```rb
+# frozen_string_literal: true
+
+class Article < ApplicationRecord
+  def self.tagged_with(tag)
+    return none unless tag.present?
+
+    joins(:taggings).where(taggings: { tag: tag }).distinct
+  end
+end
+```
+
+> You might be wondering why did we return `none` here?
+>
+> let's talk about that, generally speaking, its quite often you fiight `nil` in ruby, in this case, if we were to get `nil` back it would just be messy conditional logic where ever we used this method. So rather than dealing with that, we return an empty active record collection to cercumvent the need to deal with `nil`. This is a pretty good pattern to follow especiallywhen creating custom class methods in models.
```

Comment:
spell check, fix "fiight"; good consideration on the none. Perhaps make it clear that this would be the choice when you want chain-able methods